api/v2/performanceprofile_validation.go

```diff
@@ -0,0 +1,277 @@
+		if _, _, err := components.SplitLabelKey(k); err != nil {
+			allErrs = append(allErrs, field.Invalid(
+				field.NewPath("spec.nodeSelector"),
+				r.Spec.NodeSelector,
+				"machineConfigLabels or machineConfigPoolSelector are not set, but we  can not set it automatically because of an invalid NodeSelector label key that can't be split into domain/role"))
+		}
+	}
+
+	return allErrs
+}
+
+func (r *PerformanceProfile) validateHugePages() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if r.Spec.HugePages == nil {
+		return allErrs
+	}
+
+	// validate that default hugepages size has correct value, currently we support only 2M and 1G(x86_64 architecture)
+	if r.Spec.HugePages.DefaultHugePagesSize != nil {
+		defaultSize := *r.Spec.HugePages.DefaultHugePagesSize
+		if defaultSize != hugepagesSize1G && defaultSize != hugepagesSize2M {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.hugepages.defaultHugepagesSize"), r.Spec.HugePages.DefaultHugePagesSize, fmt.Sprintf("hugepages default size should be equal to %q or %q", hugepagesSize1G, hugepagesSize2M)))
+		}
+	}
+
+	for i, page := range r.Spec.HugePages.Pages {
+		if page.Size != hugepagesSize1G && page.Size != hugepagesSize2M {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.hugepages.pages"), r.Spec.HugePages.Pages, fmt.Sprintf("the page size should be equal to %q or %q", hugepagesSize1G, hugepagesSize2M)))
+		}
+
+		allErrs = append(allErrs, r.validatePageDuplication(&page, r.Spec.HugePages.Pages[i+1:])...)
+	}
+
+	return nil
+}
+
+func (r *PerformanceProfile) validatePageDuplication(page *HugePage, pages []HugePage) field.ErrorList {
+	var allErrs field.ErrorList
+
+	for _, p := range pages {
+		if page.Size != p.Size {
+			continue
+		}
+
+		if page.Node != nil && p.Node == nil {
+			continue
+		}
+
+		if page.Node == nil && p.Node != nil {
+			continue
+		}
+
+		if page.Node == nil && p.Node == nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.hugepages.pages"), r.Spec.HugePages.Pages, fmt.Sprintf("the page with the size %q and without the specified NUMA node, has duplication", page.Size)))
+		}
+
+		if *page.Node == *p.Node {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.hugepages.pages"), r.Spec.HugePages.Pages, fmt.Sprintf("the page with the size %q and with specified NUMA node %d, has duplication", page.Size, *page.Node)))
+		}
+	}
+
+	return allErrs
+}
+
+func (r *PerformanceProfile) validateNUMA() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if r.Spec.NUMA == nil {
+		return allErrs
+	}
+
+	// validate NUMA topology policy matches allowed values
+	if r.Spec.NUMA.TopologyPolicy != nil {
+		policy := *r.Spec.NUMA.TopologyPolicy
+		if policy != kubeletconfigv1beta1.NoneTopologyManagerPolicy &&
+			policy != kubeletconfigv1beta1.BestEffortTopologyManagerPolicy &&
+			policy != kubeletconfigv1beta1.RestrictedTopologyManagerPolicy &&
+			policy != kubeletconfigv1beta1.SingleNumaNodeTopologyManager {
```

Comment:
in a future PR we may want to avoid duplicating this validation.

Reply:
Can you please clarify?